src/monster.cpp

```diff
@@ -162,6 +163,36 @@ void monster::poly(mtype *t)
     faction = t->default_faction;
 }
 
+void monster::update_check(){
+    if (type->upgrade_group == "NULL"){
+        return;
+    }
+    int current_day = calendar::turn.get_turn()/ DAYS(1);
+    int upgrade_time = type->upgrade_min * ACTIVE_WORLD_OPTIONS["MONSTER_GROUP_DIFFICULTY"];
+    //g->u.add_msg_if_player(m_debug, "Upgrade group: %s ", type->upgrade_group);
+    g->u.add_msg_if_player(m_debug, "Current:day: %d", current_day);
```

Comment:
This `g->u.add_msg_if_player` implies some interesting philosophical problems: Is the player character really a player? If not, who *is* playing?

`g->u` is always the player character and therefor `g->u.add_msg_if_player` will always add the message. You should therefor use the global `add_msg` function instead (which behaves the same), there is no need to drag `g->u` into this.